### PR TITLE
simple head test to cheack manual completion with empty Source

### PR DESCRIPTION
**Pull request by pjazdzewski1990**
Refs https://github.com/akka/akka/issues/20182

### CODE REVIEW

**Comment by pjazdzewski1990** [comment]
I've opened a simple test case for https://github.com/akka/akka/issues/20182 cc @ktoso  @TimothyKlim

**Comment by ktoso** [comment]
OK TO TEST

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala [hunk 1]
```diff
@@ -23,6 +24,24 @@ class MethodDirectivesSpec extends RoutingSpec {
     }
   }
 
+  "head" should {
+    val headRoute = head {
+      println(s"Completing")
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: no println please

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala [hunk 2]
```diff
@@ -23,6 +24,24 @@ class MethodDirectivesSpec extends RoutingSpec {
...
+      complete(HttpEntity.Default(
+        ContentTypes.`application/octet-stream`,
+        12345L,
+        Source.empty
+      ))
+    }
+
+    "allow manual complete" in {
+      Head() ~> headRoute ~> check {
+        println(s"Status ${response._1}")
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: no println please

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala [hunk 3]
```diff
@@ -23,6 +24,24 @@ class MethodDirectivesSpec extends RoutingSpec {
...
+        status shouldEqual StatusCodes.OK
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: assert content length and actual response body?

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala (hunk 1 above)
Comment (drewhk) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: remaining printlns?

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala (hunk 2 above)
Comment (drewhk) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: println?

Comment (pjazdzewski1990) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: Too fast :) Sorry

**Comment by ktoso** [comment]
LGTM, thanks

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala [hunk 4]
```diff
@@ -23,6 +27,26 @@ class MethodDirectivesSpec extends RoutingSpec {
...
+        Source.empty
+      ))
+    }
+
+    "allow manual complete" in {
+      Head() ~> headRoute ~> check {
+        status shouldEqual StatusCodes.OK
+
+        val lengthF = response._3.dataBytes.runFold(0)((c, _) => c+1)
+        val length = Await.result(lengthF, Duration(100, "millis"))
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: FYI, usually in scala one would:

```scala
import scala.concurrent.duration._
100.millis
```

Comment (pjazdzewski1990) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala]: 👍

**Pull request merged**